src/main/java/com/onarandombox/multiverseinventories/share/Sharables.java

```diff
@@ -582,6 +590,149 @@ public boolean updatePlayer(Player player, PlayerProfile profile) {
+
+                    boolean announceAdvancements = player.getWorld().getGameRuleValue(GameRule.ANNOUNCE_ADVANCEMENTS);
+                    if (announceAdvancements) player.getWorld().setGameRule(GameRule.ANNOUNCE_ADVANCEMENTS, false);
+                    while (advancementIterator.hasNext()) {
+                        Advancement advancement = advancementIterator.next();
+                        // the set of Advancements to revoke
+                        Set<String> revoke = new HashSet<>(advancement.getCriteria());
+                        // the set of Advancements to grant
+                        Set<String> intersection = new HashSet<>(advancement.getCriteria());
+
+                        revoke.removeAll(processedAdvancements);
+                        revoke.removeAll(completedAdvancements);
+                        intersection.removeAll(processedAdvancements);
+                        intersection.retainAll(completedAdvancements);
+
+                        for (String criteria: revoke) {
+                            processedAdvancements.add(criteria);
+                            completedAdvancements.remove(criteria);
+                            player.getAdvancementProgress(advancement).revokeCriteria(criteria);
+                        }
+                        for (String criteria: intersection) {
+                            processedAdvancements.add(criteria);
+                            completedAdvancements.remove(criteria);
+                            player.getAdvancementProgress(advancement).awardCriteria(criteria);
+                        }
+
+                        // here's the idea: revoke all (criteria \ completedAdvancements), grant (criteria intersect completedAdvancements)
+                        // also, we don't need to grant/revoke anything in processedAdvancements since they've already been granted/revoked!
+                    }
+                    if (announceAdvancements) player.getWorld().setGameRule(GameRule.ANNOUNCE_ADVANCEMENTS, true);
+                    return advancements != null;
+                }
+            }).defaultSerializer(new ProfileEntry(false, "advancements")).altName("achievements").build();
+
+    /**
+     * Sharing Statistics.
+     */
+    public static final Sharable<Map> GAME_STATISTICS = new Sharable.Builder<Map>("game_statistics", Map.class,
+            new SharableHandler<Map>() {
+                @Override
+                public void updateProfile(PlayerProfile profile, Player player) {
+                    Map<String, Integer> playerStats = new HashMap<>();
+                    for (Statistic stat: Statistic.values()) {
+                        if (stat.getType() == Statistic.Type.UNTYPED) {
+                            int val = player.getStatistic(stat);
+                            // no need to save values of 0, that's the default!
+                            if (val != 0) playerStats.put(stat.name(), val);
```

Comment:
Notices some of your if/else conditions doesn't have enclosing `{}`, iirc @dumptruckman prefer with them?

Reply:
Yeah, I'll change this.